### PR TITLE
[IBM MQ] Add Integration Package with Error Log Data Stream

### PR DESCRIPTION
**Pull request by harnish-crest**
- Enhancement

## What does this PR do?

- Generated the skeleton of IBM MQ integration package.
- Added 1 data stream (Queue Manager Error Logs).
- Added data collection logic for the data streams.
- Added the ingest pipeline for the data streams.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yaml files.
- Added dashboards and visualizations.
- Added system test cases for the data stream.

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's changelog.yml file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

- Clone integrations repo.
- Install elastic-package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/ibmmq directory.
- Run the following command to run tests.
elastic-package test

## Related issues

- Relates #620
- Relates #3637

## Screenshots
IBM MQ errorlog dashboard

### CODE REVIEW

**Review by kush-elastic — CHANGES_REQUESTED** [review]
Please change docker image tag from `latest` to specific version and lets use `variants.yml` for multiple version test.

File: packages/ibmmq/_dev/deploy/docker/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+version: '2.3'
+services:
+  ibmmq:
+    image: ibmcom/mq:latest
```
[the remark this answers is not in the document]
Comment (harnish-crest) [reply packages/ibmmq/_dev/deploy/docker/docker-compose.yml]: Added vriants.yml with latest version 9.2.4.0

File: packages/ibmmq/_dev/deploy/variants.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+variants:
+  v9.2.4.0-r1-amd64:
+    SERVICE_VERSION: 9.2.4.0-r1-amd64
+default: 9.2.4.0-r1-amd64
```
Comment (kush-elastic) [inline packages/ibmmq/_dev/deploy/variants.yml:4]: nit: default will contain variants key name, which is in your case is `v9.2.4.0-r1-amd64`.
```suggestion
  v9.2.4.0-r1-amd64:
    SERVICE_VERSION: 9.2.4.0-r1-amd64
default: v9.2.4.0-r1-amd64
```

**Review by kush-elastic — APPROVED** [review]
Added a comment.
else LGTM!

**Review by kush-elastic — APPROVED** [review]
LGTM!

File: packages/ibmmq/manifest.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+name: ibmmq
+title: IBM MQ
+version: 0.1.0
+format_version: 1.0.0
+description: Collect logs and metrics from IBM MQ with Elastic Agent.
+type: integration
+categories:
+  - web
```
[the remark this answers is not in the document]
Comment (harnish-crest) [reply packages/ibmmq/manifest.yml:8]: Makes sense, thanks!

File: packages/ibmmq/_dev/build/docs/README.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+This integration has been tested against `IBM MQ v9.1` and `IBM MQ v9.2`.
+
+## Requirements
+
+In order to ingest data from IBM MQ:
+- User must know the path of IBM MQ Queue Manager Error logs. (default paths: `/var/mqm/errors/*.LOG` and `/var/mqm/qmgrs/*/errors/*.LOG`)
+
+## Logs
+
```
[the remark this answers is not in the document]
Comment (harnish-crest) [reply packages/ibmmq/_dev/build/docs/README.md:15]: Not quite sure what you are trying to refer to here. Could you please elaborate? This is the standard format of README that most of the integrations follow.

File: packages/ibmmq/_dev/build/docs/README.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+### Queue Manager Error logs
+
+The `errorlog` data stream collects Error logs of Queue Manager which include the description, action, explanation and code of the error.
```
[the remark this answers is not in the document]
Comment (harnish-crest) [reply packages/ibmmq/_dev/build/docs/README.md:18]: Not quite sure what you are trying to refer to here. Could you please elaborate? This is the standard format of README that most of the integrations follow.

[the remark this answers is not in the document]
Comment (harnish-crest) [reply packages/ibmmq/_dev/build/docs/README.md:18]: I have referred to this link. Let me know does this link look good to you?

[https://www.site24x7.com/help/log-management/ibm-mq-error-logs.html](https://www.site24x7.com/help/log-management/ibm-mq-error-logs.html)

[the remark this answers is not in the document]
Comment (ritalwar) [reply packages/ibmmq/_dev/build/docs/README.md:18]: Looks good! Please add it in the docs as well.

[the remark this answers is not in the document]
Comment (harnish-crest) [reply packages/ibmmq/_dev/build/docs/README.md]: Added it, thanks.

File: packages/ibmmq/_dev/build/docs/README.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+The IBM MQ Integration is used to fetch observability data from [IBM MQ web endpoints](https://www.ibm.com/docs/en/ibm-mq) and ingest it into Elasticsearch.
+
+## Compatibility
+
+This integration has been tested against `IBM MQ v9.1` and `IBM MQ v9.2`.
+
+## Requirements
+
+In order to ingest data from IBM MQ:
+- User must know the path of IBM MQ Queue Manager Error logs. (default paths: `/var/mqm/errors/*.LOG` and `/var/mqm/qmgrs/*/errors/*.LOG`)
```
Comment (ritalwar) [inline packages/ibmmq/_dev/build/docs/README.md:12]: 'user must know the path'. This line can be modified and it should be kind of 'user should specify the path'.

File: packages/ibmmq/data_stream/errorlog/manifest.yml [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        type: bool
+        multi: false
+        default: false
+      - name: processors
+        type: yaml
+        title: Processors
+        multi: false
+        required: false
+        show_user: false
+        description: >
```
Comment (ritalwar) [inline packages/ibmmq/data_stream/errorlog/manifest.yml:39]: can we omit > and give description directly, like in other fields to maintain consistent format?

File: packages/ibmmq/_dev/build/docs/README.md (hunk 6 above)
Comment (harnish-crest) [reply packages/ibmmq/_dev/build/docs/README.md:12]: Makes sense, Thanks.

File: packages/ibmmq/data_stream/errorlog/manifest.yml (hunk 7 above)
Comment (harnish-crest) [reply packages/ibmmq/data_stream/errorlog/manifest.yml]: Updated, thanks

File: packages/ibmmq/_dev/build/docs/README.md [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+The IBM MQ Integration is used to fetch observability data from [IBM MQ web endpoints](https://www.ibm.com/docs/en/ibm-mq) and ingest it into Elasticsearch.
+
+## Compatibility
+
+This integration has been tested against `IBM MQ v9.1` and `IBM MQ v9.2`.
+
+## Requirements
+
+In order to ingest data from IBM MQ:
+- User should specify the path IBM MQ Queue Manager Error logs. (default paths: `/var/mqm/errors/*.LOG` and `/var/mqm/qmgrs/*/errors/*.LOG`)
```
Comment (ritalwar) [inline packages/ibmmq/_dev/build/docs/README.md:12]: nit: path of IBM MQ.
Please check the docs and descriptions again and correct them.

**Review by ritalwar — APPROVED** [review]
LGTM!

**Pull request merged**